src/dune_rules/dune_file.ml

```diff
@@ -764,17 +768,62 @@ module Library = struct
   let has_foreign_cxx t = Buildable.has_foreign_cxx t.buildable
 
   let foreign_archives t =
-    (if List.is_empty t.buildable.foreign_stubs then []
-    else [ Foreign.Archive.stubs (Lib_name.Local.to_string (snd t.name)) ])
-    @ List.map ~f:snd t.buildable.foreign_archives
-
-  let foreign_lib_files t ~dir ~ext_lib =
-    List.map (foreign_archives t) ~f:(fun archive ->
-        Foreign.Archive.lib_file ~archive ~dir ~ext_lib)
+    let lib_stubs =
+      if List.is_empty t.buildable.foreign_stubs then None
+      else Some (Foreign.Archive.stubs (Lib_name.Local.to_string (snd t.name)))
+    in
+    (lib_stubs, List.map ~f:snd t.buildable.foreign_archives)
```

Comment:
Instead of returning a tuple, let's just have two separate functions. The return type is just confusing and there's no performance to gain from computing them together.

Reply:
04e68dc0534b98045ffae9ed9ec0bfb39b4f5bf9